JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/favourites/MovieFilterChipRow.kt

```diff
@@ -16,129 +16,60 @@
+                    val updated = if (it) {
+                        selectedFilterList.items + listOf(filterCondition)
+                    } else {
+                        selectedFilterList.items - setOf(filterCondition)
                     }
-                    MovieFilterChip(
-                        label = chipItem(),
-                        isChecked = isChecked,
-                        modifier = Modifier.ifElse(
-                            index == 0,
-                            focusRestorerModifiers.childModifier
-                        ),
-                        onCheckedChange = {
-                            when (chipItem) {
-                                ChipListItems.Movies -> {
-                                    isMovieFilterSelected = !isMovieFilterSelected
-                                }
-
-                                ChipListItems.TV_Shows -> {
-                                    isTvShowsFilterSelected = !isTvShowsFilterSelected
-                                }
-
-                                ChipListItems.Added_Last_Week -> {
-                                    isAddedLastWeekFilterSelected = !isAddedLastWeekFilterSelected
-                                }
-
-                                ChipListItems.Available_in_4K -> {
-                                    isAvailableIn4KFilterSelected = !isAvailableIn4KFilterSelected
-                                }
-                            }
-                        }
-                    )
+                    onSelectedFilterListUpdated(FilterList(updated))
+                },
+                modifier = if (index == 0) {
+                    focusRestorerModifiers.childModifier
+                } else {
+                    Modifier
                 }
-            }
-        }
-    }
-
-    LaunchedEffect(isMovieFilterSelected) {
-        if (isMovieFilterSelected) {
-            movieListRange.addAll(movieFilterRange)
-        } else {
-            movieListRange.removeAll(movieFilterRange)
-        }
-    }
-
-    LaunchedEffect(isTvShowsFilterSelected) {
-        if (isTvShowsFilterSelected) {
-            movieListRange.addAll(tvShowsFilterRange)
-        } else {
-            movieListRange.removeAll(tvShowsFilterRange)
-        }
-    }
-
-    LaunchedEffect(isAddedLastWeekFilterSelected) {
-        if (isAddedLastWeekFilterSelected) {
-            movieListRange.addAll(addedLastWeekFilterRange)
-        } else {
-            movieListRange.removeAll(addedLastWeekFilterRange)
+            )
         }
     }
+}
 
-    LaunchedEffect(isAvailableIn4KFilterSelected) {
-        if (isAvailableIn4KFilterSelected) {
-            movieListRange.addAll(availableIn4KFilterRange)
-        } else {
-            movieListRange.removeAll(availableIn4KFilterRange)
-        }
+fun filterLabelId(filterCondition: FilterCondition): Int {
+    return when (filterCondition) {
+        FilterCondition.Movies -> R.string.favorites_movies
```

Comment:
If FilterCondition was an enum it could have a resource id as paramanter:

```enum class FilterCondition(val idList: List<Int>, @StringRes labelResId: Int)```

Reply:
Thanks. That is a good suggestion to make code more declarative.